### ISSUE TITLE
Connection Pool throws ArrayIndexOutOfBoundsException when try to create connection

### ISSUE THREAD

**Issue by cyhii** [eclipse-vertx/vertx-sql-client#1136]
### Questions

My application started to serve requests for a while, everything works fine. In the next few hours no request comes in, after that, when a new request received, the application is broken. Here is the stack trace:

```
java.lang.ArrayIndexOutOfBoundsException: 100
	at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1.apply(SqlConnectionPool.java:92)
	at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1.apply(SqlConnectionPool.java:88)
	at io.vertx.core.net.impl.pool.SimpleConnectionPool$Acquire.execute(SimpleConnectionPool.java:567)
	at io.vertx.core.net.impl.pool.SimpleConnectionPool$Acquire.execute(SimpleConnectionPool.java:535)
	at io.vertx.core.net.impl.pool.CombinerExecutor.pollAndExecute(CombinerExecutor.java:58)
	at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:44)
	at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:240)
	at io.vertx.core.net.impl.pool.SimpleConnectionPool.acquire(SimpleConnectionPool.java:620)
	at io.vertx.core.net.impl.pool.SimpleConnectionPool.acquire(SimpleConnectionPool.java:624)
	at io.vertx.sqlclient.impl.pool.SqlConnectionPool.execute(SqlConnectionPool.java:148)
	at io.vertx.sqlclient.impl.PoolImpl.schedule(PoolImpl.java:163)
	at io.vertx.sqlclient.impl.QueryExecutor.executeExtendedQuery(QueryExecutor.java:157)
	at io.vertx.sqlclient.impl.SqlClientBase$PreparedQueryImpl.execute(SqlClientBase.java:155)
	at io.vertx.sqlclient.impl.SqlClientBase$PreparedQueryImpl.execute(SqlClientBase.java:160)
	at io.vertx.sqlclient.impl.SqlClientBase$PreparedQueryImpl.execute(SqlClientBase.java:166)
```

It seems that connections in the pool has been released, so the pool try to create new connection. May be the ContextProvider should use a more safe index to fetch EventLoop from loops array?

My PoolOptions like this:

```kotlin
val poolOptions = PoolOptions()
      .setMaxSize(config.getString("mysql.pool.max-size", "20").toInt())
      .setEventLoopSize(config.getString("mysql.pool.event-loop-size", "1").toInt())
      .setName("mysql-pool")
```

I've not set the `idleTimeout` option, but the connection be released?

### Version

4.2.2

**Comment by cyhii** [comment]
Already changed to use `io.vertx:vertx-jdbc-client` to prevent this bug. Anyone who met this same issue can also have a try.

**Comment by vietj** [comment]
this is a terrible bug, it will be fixed in 4.2.5 released this week.

in the meantime you can disable the eventLoopSize setting on the client and it will work as usual
